### PR TITLE
Handle libxml errors and warnings

### PR DESCRIPTION
**Pull request by maxammann**
This PR improves how libxml errors are handled. Because there can be a lot of duplicate errrors I decided to count similar errors and show how often it occurred. The log output for multiple failure looks like this:

```
{"reqId":"EYcnDGGmhWZ8kXC4Te7E","level":2,"time":"2020-10-25T18:02:51+00:00","remoteAddr":"2a02:810d:640:1e52:3f8c:68c2:b41d:b462","user":"max","app":"no app in context","method":"POST","url":"/index.php/apps/cookbook/import","message":"libxml: Error 68 occurred 907 times while parsing https://www.chefkoch.de/rezepte/18711004787789/Risotto-mit-gruenem-Spargel-und-Parmesan.html. Last time in line 7728 and column 119: htmlParseEntityRef: no name\n","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:81.0) Gecko/20100101 Firefox/81.0","version":"19.0.4.2"}
{"reqId":"EYcnDGGmhWZ8kXC4Te7E","level":2,"time":"2020-10-25T18:02:51+00:00","remoteAddr":"2a02:810d:640:1e52:3f8c:68c2:b41d:b462","user":"max","app":"no app in context","method":"POST","url":"/index.php/apps/cookbook/import","message":"libxml: Error 801 occurred 329 times while parsing https://www.chefkoch.de/rezepte/18711004787789/Risotto-mit-gruenem-Spargel-und-Parmesan.html. Last time in line 10225 and column 50: Tag amp-install-serviceworker invalid\n","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:81.0) Gecko/20100101 Firefox/81.0","version":"19.0.4.2"}
{"reqId":"EYcnDGGmhWZ8kXC4Te7E","level":2,"time":"2020-10-25T18:02:51+00:00","remoteAddr":"2a02:810d:640:1e52:3f8c:68c2:b41d:b462","user":"max","app":"no app in context","method":"POST","url":"/index.php/apps/cookbook/import","message":"libxml: Error 42 occurred 3 times while parsing https://www.chefkoch.de/rezepte/18711004787789/Risotto-mit-gruenem-Spargel-und-Parmesan.html. Last time in line 5041 and column 109: Attribute parentid redefined\n","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:81.0) Gecko/20100101 Firefox/81.0","version":"19.0.4.2"}
{"reqId":"EYcnDGGmhWZ8kXC4Te7E","level":2,"time":"2020-10-25T18:02:51+00:00","remoteAddr":"2a02:810d:640:1e52:3f8c:68c2:b41d:b462","user":"max","app":"no app in context","method":"POST","url":"/index.php/apps/cookbook/import","message":"libxml: Error 23 occurred 333 times while parsing https://www.chefkoch.de/rezepte/18711004787789/Risotto-mit-gruenem-Spargel-und-Parmesan.html. Last time in line 10210 and column 96: htmlParseEntityRef: expecting ';'\n","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:81.0) Gecko/20100101 Firefox/81.0","version":"19.0.4.2"}
{"reqId":"EYcnDGGmhWZ8kXC4Te7E","level":2,"time":"2020-10-25T18:02:51+00:00","remoteAddr":"2a02:810d:640:1e52:3f8c:68c2:b41d:b462","user":"max","app":"no app in context","method":"POST","url":"/index.php/apps/cookbook/import","message":"libxml: Error 76 occurred 10 times while parsing https://www.chefkoch.de/rezepte/18711004787789/Risotto-mit-gruenem-Spargel-und-Parmesan.html. Last time in line 6509 and column 7: Unexpected end tag : div\n","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:81.0) Gecko/20100101 Firefox/81.0","version":"19.0.4.2"}
```

The problem with the handling before this is that nextcloud alwas logs the parameters of the function where an libxml error occurred. This meany in the above example the $html is logged a few hundred times.

### CODE REVIEW

**Comment by christianlupus** [comment]
First of all: Thank you for contributing to the app!

For the record: This should solve #344.

Two comments on your contribution, @maxammann:

1. You should signoff your commit. Depending on your local envorinmant this is done by `git commit -s`. As only one commit has been made, you might be able to `git commit --amend -s && git push --force-with-lease`. If you are using a GUI of your IDE you will have to look up how this needs to be used.
2. Just a hint for future work: You had many changes of empty lines. In fact you removed the indentations of these lines. Of course, this introduces no change in the functionality but might cause merge conflicts with other peoples' work. If your IDE is doing this "behind your back", you might want to add only the relevant lines to your commits to keep the history of the git repository cleaner.

Thank you very much.

**Comment by maxammann** [comment]
@christianlupus thanks for the heads up :) changes for new lines have been reverted and I signed that commit off. There's still some testing I need to do

File: composer.json [hunk 1]
```diff
@@ -8,7 +8,9 @@
             "name": "Jeppe Zapp"
         }
     ],
-    "require": {},
+    "require": {
+      "ext-libxml": "*"
```
Comment (maxammann) [inline composer.json:12]: I included this as IntelliJ told me so. Some PHP expert has to decide whether this is fine, or good.

Comment (christianlupus) [reply composer.json:12]: This look pretty fine to me after a quick research. I'd stick with that.

**Comment by christianlupus** [comment]
All in all, this looks good to me. I will merge it and in case any issues arise, we might solve the bug then.

**Pull request merged**